datadog/resource_datadog_dashboard.go

```diff
@@ -122,9 +149,47 @@ func resourceDatadogDashboardUpdate(d *schema.ResourceData, meta interface{}) er
 	if _, _, err = datadogClientV1.DashboardsApi.UpdateDashboard(authV1, id).Body(*dashboard).Execute(); err != nil {
 		return translateClientError(err, "error updating dashboard")
 	}
+
+	updateDashboarLists(d, providerConf, *dashboard.Id)
+
 	return resourceDatadogDashboardRead(d, meta)
 }
 
+func updateDashboarLists(d *schema.ResourceData, providerConf *ProviderConfiguration, dashboardId string) {
+	dashTypeString := "custom_screenboard"
+	if d.Get("layout_type").(string) == "free" {
+		dashTypeString = "custom_timeboard"
```

Comment:
shouldn't it be the opposite ? `custom_screenboard` for `free` and `custom_timeboard` otherwise ?

Reply:
It's very possible. I think the backend corrects anyway, but I'm always confused by terminology.